### PR TITLE
providers/libvirt: support for using eth0 as iface

### PR DESCRIPTION
**Pull request by jforman**
running coreos on libvirt (ubuntu) on bare-metal, the default
network interface is eth0, not eth1 as in use by other
bare-metal providers (vagrant-virtualbox). This provider mirrors
that provider almost exactly, just s/eth1/eth0/g.

### CODE REVIEW

File: src/providers/libvirt/mod.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+use hostname;
+use pnet;
+
+pub fn fetch_metadata() -> Result<Metadata> {
+    let h = hostname::get_hostname().ok_or("unable to get hostname")?;
+    let ip = get_ip()?;
+
+    Ok(Metadata::builder()
+       .add_attribute("LIBVIRT_PRIVATE_IPV4".to_owned(), ip)
+       .add_attribute("LIBVIRT_HOSTNAME".to_owned(), h.clone())
```
Comment (lucab) [inline src/providers/libvirt/mod.rs:34]: `COREOS_LIBVIRT_HOSTNAME`

File: src/providers/libvirt/mod.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+use hostname;
+use pnet;
+
+pub fn fetch_metadata() -> Result<Metadata> {
+    let h = hostname::get_hostname().ok_or("unable to get hostname")?;
+    let ip = get_ip()?;
+
+    Ok(Metadata::builder()
+       .add_attribute("LIBVIRT_PRIVATE_IPV4".to_owned(), ip)
```
Comment (lucab) [inline src/providers/libvirt/mod.rs:33]: `COREOS_LIBVIRT_IPV4_LOCAL`

**Review by lucab — CHANGES_REQUESTED** [review]

**Comment by lucab** [comment]
Naming aside, I'm not really sure we should introduce this new `libvirt` platform. It doesn't really use any externally-sourced metadata, just transient machine state.
libvirt has a dedicated [metadata section](https://libvirt.org/formatdomain.html#elementsMetadata) but the two are not guaranteed to match (and I think there is no way to introspect that from the guest).

I think `vagrant-virtualbox` is the odd one here for historical reasons and we are still addressing that in several places. I won't introduce more of that under a new platform.

**Comment by jforman** [comment]
Okay, thanks Luca, that all sounds reasonable. Could you point me to where those vagrant-virtualbox 'historical reasons' are being addressed? I'd like to follow along.

**Comment by lucab** [comment]
I don't have a single pointer for that, but there a few reference to chase starting from https://github.com/coreos/bugs/issues/2264#issuecomment-346582204.

**Pull request closed without merging**